File: FSharp/Implementation/Day1.fs
namespace Implementation

open System

type Day1 =
    
    static member Part1 (input : string) =
        let notNullOrWhiteSpace p = not (String.IsNullOrWhiteSpace p) 
        input.Split '\n'
        |> Seq.ofArray
        |> Seq.filter notNullOrWhiteSpace
        |> Seq.map Int32.Parse
        |> Seq.pairwise
        |> Seq.filter (fun (a, b) -> a < b)
        |> Seq.length
    
    static member Part2 (input : string) =
        let notNullOrWhiteSpace p = not (String.IsNullOrWhiteSpace p) 
        input.Split '\n'
        |> Seq.ofArray
        |> Seq.filter notNullOrWhiteSpace
        |> Seq.map Int32.Parse
        |> Seq.map float
        |> Seq.windowed 3
        |> Seq.map Array.average
        |> Seq.pairwise
        |> Seq.filter (fun (a, b) -> a < b)
        |> Seq.length

File: CSharp/Implementation/Day1.cs
namespace Implementation;

public static class Day1
{
    public static int Part1(string input)
    {
        return 
            input
                .Split('\n')
                .Where(x => !string.IsNullOrWhiteSpace(x))
                .Select(int.Parse)
                .Aggregate((0, int.MaxValue),
                    (state, val) => 
                        val > state.MaxValue 
                            ? (state.Item1 + 1, val)
                            : (state.Item1, val))
                .Item1;
    }
}
